### PR TITLE
Avoid nil pointer de-reference panic when accidentally passing nil into WithError() on zerolog

### PR DESCRIPTION
**Pull request by acgreek**
We had a case where someone accidentally passed nil as the error value into WithError. This caused a panic with in zerolog Err() method.

### CODE REVIEW

File: zerolog/zerolog.go [hunk 1]
```diff
@@ -279,8 +284,16 @@ func (e *entry) WithError(errs ...error) log.Entry {
 	}
 
 	if le == 1 {
+		if errs[0] == nil {
+			errs[0] = nilError
```
Comment (iaburton) [inline zerolog/zerolog.go:288]: So the stdlib when passing something nil to a printer might just do something like `<nil>` to indicate nil was passed, but since this method is specifically for tracking errors I wonder if we should just return when a nil error is passed? Rather than adding an error _about a nil error_

Comment (acgreek) [reply zerolog/zerolog.go:288]: I prefer to be told that I am passing nil to WithError rather than ignore the the fact that we made this mistake

Comment (hews) [reply zerolog/zerolog.go:288]: I like this solution, will wait for @iaburton to weigh in.

Comment (hews) [reply zerolog/zerolog.go:288]: Would like to see a test case that refers directly to this behavior though.

**Comment by hews** [comment]
OK, after talking with @iaburton my final take is:

1. We need a test case for the behavior of drivers when we pass a `nil` to `Entry.WithError`.
2. We need all the drivers to act the same, and the default behavior should be to just ignore nil errors. The reason for this is that it would likely cause a lot of these errors to start showing up in DD and then everyone will yell at us about the logger again.
3. We can add an option that allows us to use a `ErrNilPassedToWithError` (which should be the same for each driver) when a nil is passed.

Sorry this is a significantly bigger change, but again we are seeing an issue where the solution would cause the `zerolog` and `logrus` drivers to diverge further, which we want to avoid.

**Comment by hews** [comment]
Superseded by #22.
